integration-libs/opf/base/components/opf-quick-buy/google-pay/google-pay.service.spec.ts

```diff
@@ -4,4 +4,693 @@
+      expect(container.nativeElement.children[0].tagName).toBe('BUTTON');
+      expect(mockGooglePaymentClient.createButton).toHaveBeenCalled();
+    });
+
+    it('should attach the correct click handler to the button', () => {
+      const container = new ElementRef(document.createElement('div'));
+      spyOn(service, 'initTransaction');
+
+      service.renderPaymentButton(container);
+
+      const button = container.nativeElement.children[0];
+      (button as any).click();
+
+      expect(service.initTransaction).toHaveBeenCalled();
+    });
+  });
+
+  describe('handlePaymentCallbacks', () => {
+    const encodedMockToken = 'encodedMockToken';
+
+    beforeEach(() => {
+      spyOn(window, 'btoa').and.callFake(() => {
+        return encodedMockToken;
+      });
+    });
+
+    it('should return valid payment data callbacks', () => {
+      const callbacks = service.handlePaymentCallbacks();
+
+      expect(callbacks).toBeDefined();
+      expect(callbacks.onPaymentAuthorized).toBeDefined();
+      expect(callbacks.onPaymentDataChanged).toBeDefined();
+    });
+
+    describe('onPaymentAuthorized', () => {
+      it('should handle payment authorization', (done) => {
+        const callbacks = service.handlePaymentCallbacks();
+        const mockCartId = 'cartId';
+        const mockToken = 'mockToken';
+        const paymentDataResponse = {
+          paymentMethodData: {
+            tokenizationData: {
+              token: mockToken,
+            },
+          },
+        } as google.payments.api.PaymentData;
+
+        mockCartHandlerService.getCurrentCartId.and.returnValue(of(mockCartId));
+        mockPaymentFacade.submitPayment.and.returnValue(of(true));
+        mockCartHandlerService.setDeliveryAddress.and.returnValue(
+          of('addressId')
+        );
+
+        if (callbacks.onPaymentAuthorized) {
+          (
+            callbacks.onPaymentAuthorized(paymentDataResponse) as Promise<any>
+          ).then((result) => {
+            const submitPaymentArgs =
+              mockPaymentFacade.submitPayment.calls.mostRecent().args[0];
+
+            expect(result).toBeDefined();
+            expect(mockPaymentFacade.submitPayment).toHaveBeenCalled();
+            expect(submitPaymentArgs.callbackArray.length).toBe(3);
+            submitPaymentArgs.callbackArray.forEach(
+              (callback: () => Promise<void>) => {
+                expect(typeof callback).toBe('function');
+              }
+            );
+            expect(submitPaymentArgs.cartId).toBe(mockCartId);
+            expect(submitPaymentArgs.cartId).toBe(mockCartId);
+            expect(submitPaymentArgs.encryptedToken).toBe(encodedMockToken);
+            expect(submitPaymentArgs.paymentMethod).toBe(
+              PaymentMethod.GOOGLE_PAY
+            );
+            expect(mockCartHandlerService.getCurrentCartId).toHaveBeenCalled();
+            expect(result).toEqual({ transactionState: 'SUCCESS' });
```

Comment:
@FollowTheFlo Hello, Florent. Want to consult this. Seems like result always will be `transactionState: 'SUCCESS'`.

In the service, in line 253 the result has type 
```
{
    submitPayment: boolean;
}
```

In line 254 we have assignment `const isSuccess = result`
so isSuccess will also have type 
```
{
    submitPayment: boolean;
}
```

Thats mean that in this check `isSuccess ? 'SUCCESS' : 'ERROR'` because `isSuccess` is object, it will return 'SUCCESS' every time.

Seems like here `const isSuccess = result` should be `const isSuccess = result.submitPayment`. 

Is it make sense for you? I'm not sure because I haven't worked on the implementation, so I need your consultation. Thank you

Reply:
I think that for now we need to mock always the "SUCCESS" status as we are working on the testing environment, and covered only a happy path scenario. So in this case such check looks ok.